test/decl/typealias/generic.swift

```diff
@@ -89,7 +89,7 @@ let _ : D<Int, Int, Float> = D(a: 1, b: 2)
 let _ : F = { (a : Int) -> Int in a }  // Infer the types of F
 
 // TODO QoI: Cannot infer T1/T2.
-let _ : F = { a in a }  // expected-error {{cannot convert value of type '(_) -> _' to specified type 'F'}}
+let _ : F = { a in a }  // expected-error {{type of expression is ambiguous without more context}}
```

Comment:
I saw that and I think we need to bring diagnostics of the closures to the whole new level, probably the way to go about improving at least inference diagnostic is going to be trying to type check parameters and result type allowing to produce unresolved types and then running diagnostics against them, I've created https://bugs.swift.org/browse/SR-3493 to address it.